cpp/build/Makefile

```diff
@@ -60,6 +60,9 @@ endif
 else ifeq ($(UNAME),NetBSD)
 CFLAGS += -I/usr/pkg/include/libusb-1.0
 LDFLAGS+= -shared -L/usr/pkg/lib -lusb-1.0 -Wl,-rpath=/usr/pkg/lib
+else ifeq ($(UNAME),SunOS)
+CFLAGS+=  -I$(PREFIX)/include/libusb-1.0
```

Comment:
This is the only part of the SunOS specific code that requires the PREFIX (or `/opt/local`) variable to be set.  There are a couple alternative approaches that we could use here - I'm open to any of them as I don't personally know which one is preferred or "best".

### 1. Use `pkg-config`.

`pkg-config` is an optional dependency AFAICT with this project so I'd hate to enforce it's presence - though it is incredibly useful in this specific situation.

```
[root@test ~]# uname -a
SunOS test.rapture.com 5.11 joyent_20180510T153535Z i86pc i386 i86pc Solaris
[root@test ~]# pkg-config libusb-1.0 --cflags
-I/opt/local/include/libusb-1.0
```

That will give us the CFLAGS line we need for libusb-1.0

### 2. Use a relative `#include`

Inside `cpp/hidapi/libusb/hid.c`, the include line could be modified to be:

``` c
#if defined(__sun)
#include "libusb-1.0/libusb.h"
#else
#include "libusb.h"
#endif
```

The bonus here is that we could also add `__NetBSD__` to the `defined()` check to the drop it's custom CFLAGS in the lines above.

### 3. Using `$PREFIX`

This is what is currently used in this PR